synapse/http/matrixfederationclient.py

```diff
@@ -706,8 +706,13 @@ async def put_json(
             timeout=timeout,
         )
 
+        if timeout is not None:
+            _sec_timeout = timeout / 1000
+        else:
+            _sec_timeout = self.default_timeout
+
         body = await _handle_json_response(
-            self.reactor, self.default_timeout, request, response, start_ms
+            self.reactor, _sec_timeout, request, response, start_ms
```

Comment:
yeah, it's definitely not matching the current docstring. The problem is that I don't think either the current behaviour, or the docstring, are actually sensible: what you really want is an overall timeout, rather than a timeout which applies twice in succession.

That's a rabbithole I didn't really want to get into, hence just making the methods consistent with each other. I'll have a go at updating the docstrings too.